src/Symfony/Component/String/Inflector/FrenchInflector.php

```diff
@@ -108,7 +108,7 @@ final class FrenchInflector implements InflectorInterface
      * A list of words which should not be inflected.
      * This list is only used by singularize.
      */
-    private const UNINFLECTED = '/^(abcès|accès|abus|albatros|anchois|anglais|autobus|bois|brebis|carquois|cas|chas|colis|concours|corps|cours|cyprès|décès|devis|discours|dos|embarras|engrais|entrelacs|excès|fils|fois|gâchis|gars|glas|héros|intrus|jars|jus|kermès|lacis|legs|lilas|marais|mars|matelas|mépris|mets|mois|mors|obus|os|palais|paradis|parcours|pardessus|pays|plusieurs|poids|pois|pouls|printemps|processus|progrès|puits|pus|rabais|radis|recors|recours|refus|relais|remords|remous|rictus|rhinocéros|repas|rubis|sans|sas|secours|sens|souris|succès|talus|tapis|tas|taudis|temps|tiers|univers|velours|verglas|vernis|virus)$/i';
+    private const UNINFLECTED = '/^(abcès|accès|abus|albatros|anchois|anglais|autobus|bois|brebis|carquois|cas|chas|colis|concours|corps|cours|cyprès|décès|devis|discours|dos|embarras|engrais|entrelacs|excès|fils|fois|gâchis|gars|glas|héros|intrus|jars|jus|kermès|lacis|legs|lilas|marais|mars|matelas|mépris|mets|mois|mors|obus|os|palais|paradis|parcours|pardessus|pays|plusieurs|poids|pois|pouls|printemps|processus|progrès|puits|pus|rabais|radis|recors|recours|refus|relais|remords|remous|rictus|rhinocéros|repas|rubis|sans|sas|secours|sens|souris|succès|talus|tapis|tas|taudis|temps|tiers|univers|velours|verglas|vernis|virus|albanais|angolais|bangladais|bostwanais|burundais|camerounais|congolais|finlandais|français|gabonais|guyanais|irlandais|islandais|japonais|libanais|malais|maltais|martiniquais|népalais|néo-zélandais|ougandais|pakistanais|néerlandais|polonais|portugais|réunionnais|rwandais|sénégalais|sierra-léonais|soudanais|sri lankais|taïwanais|thaïlandais|togolais|québécois|chinois|suèdois|hongrois|danois|luxembourgeois|béninois|seychellois)$/i';
```

Comment:
But this class does not handle adjectives (and that's a good thing because the concept of singular/plural for adjectives is really locale dependant)

```
/**
 * French inflector.
 *
 * This class does only inflect nouns; not adjectives nor composed words like "soixante-dix".
 */
final class FrenchInflector implements InflectorInterface
{
```

And i realize your pluralize exceptions are only adjectives, right ? 


--- 


Concerning the coutry lists, in a quick look i got : 

**Not a country:** 
- guyanais
- quebecois
- réunionnais
- ... and some other but i'm not in a mood for such debate haha

Missing:
- Azerbaïdjanais
- Bhoutanais
- Dominiquais
- Écossais
- Kazakhstanais
- Nord-Irlandais
- ... and some there too are up to a debate i will not have